src/components/Quote.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Calculator from './Calculator';
+
+const Quote = ({ category }) => {
+  const [quote, setQuote] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    // make the HTTP request to the API
+    const url = `https://api.api-ninjas.com/v1/quotes?category=${category}`;
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {
+        // update the state with the first quote
+        setQuote(data[0].quote);
+        setLoading(true);
+        setError(true);
+      })
+      .catch(() => {
+        setLoading(true);
+        setError(false);
+      });
+  }, [category]);
```

Comment:
[OPTIONAL] This is optional but recommended so please don't skip this requirement. Well done :clap: you did a very nice job :+1: but I recommended using `async` and `await` because the purpose of `async / await` is to simplify the syntax necessary to consume promise-based APIs. The behaviour of `async / await` is similar to combining generators and promises. `Async` functions always return a promise. Also, create a function and call that function in the `useEffect` then your code looks clean :+1: check below :point_down:  my suggestion :slightly_smiling_face: 

```suggestion
useEffect(() => {
  const fetchQuote = async () => {
    try {
      const url = `https://api.api-ninjas.com/v1/quotes?category=${category}`;
      const response = await fetch(url, {
        headers: {
          'X-Api-Key': 'ADD_YOUR_API_KEY_HERE'
        },
      });
      const data = await response.json();

      if (data.length > 0) {
        setQuote(data[0].quote);
      }
      
      setLoading(true);
      setError(false);
    } catch (error) {
      setLoading(true);
      setError(true);
    }
  };

  fetchQuote();
}, [category]);
```

**_Explanation of the changes:_**

- The `fetch` request is now handled using `async/await` syntax, which makes the code more readable and easier to follow.
- The logic to update the state with the first quote is now placed within an `if` condition to check if `data` has at least one element.
- The catch block handles any errors that occur during the fetch request and sets the `setError` state to `true`.
- The refactored code reduces unnecessary repetition by setting `setLoading(true)` outside of the `if` condition to ensure it is always executed.